src/v/transform/logging/tests/log_manager_test.cc

```diff
@@ -0,0 +1,273 @@
+#include "base/units.h"
+#include "config/configuration.h"
+#include "model/namespace.h"
+#include "model/transform.h"
+#include "strings/utf8.h"
+#include "test_utils/async.h"
+#include "transform/logging/event.h"
+#include "transform/logging/io.h"
+#include "transform/logging/log_manager.h"
+#include "transform/logging/tests/utils.h"
+
+#include <seastar/core/manual_clock.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include <gtest/gtest.h>
+
+#include <memory>
+
+namespace transform::logging {
+
+namespace {
+
+using namespace std::chrono_literals;
+using manager_t = transform::logging::manager<ss::manual_clock>;
+
+std::chrono::milliseconds flush_interval() {
+    return config::shard_local_cfg()
+      .data_transforms_logging_flush_interval_ms();
+}
+
+void advance_clock(ss::manual_clock::duration dur = flush_interval()) {
+    ss::manual_clock::advance(dur);
+    tests::drain_task_queue().get();
+}
+
+class fake_client : public transform::logging::client {
+public:
+    fake_client() = default;
+
+    ss::future<>
+    write(model::transform_name_view, ss::chunked_fifo<iobuf> events) override {
+        std::move(events.begin(), events.end(), std::back_inserter(_logs));
+        return ss::now();
+    }
+
+    const ss::chunked_fifo<iobuf>& logs() const { return _logs; }
+
+private:
+    ss::chunked_fifo<iobuf> _logs;
+};
+
+} // namespace
+
+class TransformLogManagerTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        auto sink = std::make_unique<fake_client>();
+        _client = sink.get();
+        _manager = std::make_unique<manager_t>(
+          model::node_id{0}, std::move(sink));
+        start_manager();
+    }
+
+    void SetUp(size_t buffer_capacity, size_t line_max) {
+        config::shard_local_cfg()
+          .data_transforms_logging_buffer_capacity_bytes.set_value(
```

Comment:
Let's use `config::mock_property` instead of mutating global state.

Reply:
🔥 
wasn't aware of that

Reply:
done